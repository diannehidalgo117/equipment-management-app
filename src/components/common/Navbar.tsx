import { Link, useLocation } from "react-router-dom";

const Navbar = () => {
  const currentPage = useLocation();

  const knownRoutes = ["/", "/home", "/register", "/detail", "/edit"];

  const isKnownRoute = knownRoutes.some(
    (route) =>
      currentPage.pathname === route ||
      (route !== "/" && currentPage.pathname.startsWith(route + "/"))
  );

  return (
    <nav className="bg-gray-800 text-white shadow-md">
      <div className="container mx-auto px-4 py-3">
        <div className="flex justify-between items-center">
          <Link to="/" className="text-xl font-bold">
            備品管理システム
          </Link>
          <div className="space-x-4">
            {isKnownRoute && (
              <Link
                to="/register"
                className="px-4 py-2 text-sm text-gray-900 bg-slate-400
              hover:bg-slate-600 hover:text-white rounded-md transition-colors"
              >
                新規登録
              </Link>
            )}
          </div>
        </div>
      </div>
    </nav>
  );
};

export default Navbar;
