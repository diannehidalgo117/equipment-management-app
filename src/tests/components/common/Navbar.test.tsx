import { render, screen } from "@testing-library/react";
import { describe, test as it, expect } from "vitest";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import Navbar from "../../../components/common/Navbar";

// BrowserRouterでラップするヘルパー関数
const renderWithRouter = (ui: React.ReactElement) => {
  return render(ui, { wrapper: BrowserRouter });
};

describe("Navbarコンポーネント", () => {
  it("アプリタイトルが表示されることを確認", () => {
    renderWithRouter(<Navbar />);

    // アプリケーションのタイトルが表示されていることを確認
    expect(screen.getByText("備品管理システム")).toBeInTheDocument();
  });

  it("ナビゲーションリンクが正しく表示されることを確認", () => {
    renderWithRouter(<Navbar />);

    // 新規登録リンクが存在することを確認
    expect(screen.getByText("新規登録")).toBeInTheDocument();
  });

  it("ナビゲーションリンクをクリックすると正しいパスに遷移することを確認", async () => {
    renderWithRouter(<Navbar />);
    const user = userEvent.setup();

    // 新規登録リンクをクリック
    const registerLink = screen.getByText(/新規登録/i);
    await user.click(registerLink);

    // location.pathnameを確認するには、モックまたは実際のルーティングテストが必要
    // この例ではリンクのhref属性を確認する簡易的な方法を示します
    expect(registerLink.closest("a")).toHaveAttribute("href", "/register");
  });

  it("スナップショットと一致することを確認", () => {
    const { asFragment } = renderWithRouter(<Navbar />);

    // コンポーネントのレンダリングが一貫していることを確認
    expect(asFragment()).toMatchSnapshot();
  });
});
