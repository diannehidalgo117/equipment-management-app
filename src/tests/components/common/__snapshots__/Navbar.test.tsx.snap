// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbarコンポーネント > スナップショットと一致することを確認 1`] = `
<DocumentFragment>
  <nav
    class="bg-gray-800 text-white shadow-md"
  >
    <div
      class="container mx-auto px-4 py-3"
    >
      <div
        class="flex justify-between items-center"
      >
        <a
          class="text-xl font-bold"
          data-discover="true"
          href="/"
        >
          備品管理システム
        </a>
        <div
          class="space-x-4"
        >
          <a
            class="px-4 py-2 text-sm text-gray-900 bg-slate-400
              hover:bg-slate-600 hover:text-white rounded-md transition-colors"
            data-discover="true"
            href="/register"
          >
            新規登録
          </a>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
